code/modules/atmospherics/gasmixtures/auxgm.dm

```diff
@@ -124,7 +124,19 @@ GLOBAL_DATUM_INIT(gas_data, /datum/auxgm, new)
 	appearance_flags = TILE_BOUND
 	vis_flags = NONE
 
-/obj/effect/overlay/gas/New(state, alph)
+//austation begin -- added 3 new args and vars
+/obj/effect/overlay/gas/New(state, alph, custom_icon, _pixel_x, _pixel_y, _flags, _layer)
 	. = ..()
 	icon_state = state
 	alpha = alph
+	if(custom_icon)
+		icon = custom_icon
+	if(_pixel_x)
+		pixel_x = _pixel_x
+	if(_pixel_y)
+		pixel_y = _pixel_y
+	if(_flags)
+		appearance_flags = _flags
+	if(_layer)
+		layer = _layer
```

Comment:
They'll be null anyway, may as well save the performance
```suggestion
	icon = custom_icon
	pixel_x = _pixel_x
	pixel_y = _pixel_y
	appearance_flags = _flags
	layer = _layer
```

Reply:
no, we don't want them to be null if it's null, then it'll make it null for every gas that we want the default thing for